docker/Dockerfile

```diff
@@ -62,35 +62,34 @@ ARG ONEAPI_INSTALL_DIR=/opt/intel/oneapi
 # Bump it if necessary.
 
 ARG CMAKE_VERSION=3.27
-ARG CMAKE_VERSION_BUILD=0
+ARG CMAKE_VERSION_BUILD=6
 
 
 # Versions of the intel python packages
 
-ARG DPCTL_GIT_BRANCH=0.14.5
+ARG DPCTL_GIT_BRANCH=0.15.1dev0
 ARG DPCTL_GIT_URL=https://github.com/IntelPython/dpctl.git
 
-ARG DPNP_GIT_BRANCH=0.12.1
+ARG DPNP_GIT_BRANCH=0.13.1dev0
 ARG DPNP_GIT_URL=https://github.com/IntelPython/dpnp.git
 
 # `numba_dpex` often requires to install from arbitrary git commits, so the
 # full history of the given branch will be cloned (as opposed to only fetching
 # files for the most recent ref), and arbirary references on this branch can be
 # selected.
-ARG NUMBA_DPEX_GIT_BRANCH=main
-ARG NUMBA_DPEX_GIT_REF=4d4fc12b80423f7c1f9c845faeba5a3438464190
+ARG NUMBA_DPEX_GIT_BRANCH=0.22.0dev0
+ARG NUMBA_DPEX_GIT_REF=0.22.0dev0
 ARG NUMBA_DPEX_GIT_URL=https://github.com/IntelPython/numba-dpex.git
 
 ARG DPCPP_LLVM_SPIRV_GIT_BRANCH=main
 ARG DPCPP_LLVM_SPIRV_GIT_URL=https://github.com/IntelPython/dpcpp-llvm-spirv/
 
-
 # Version of other python packages explicitly installed either within the
 # build environment or the runtime environment.
 
 ARG NUMPY_VERSION="<1.25"
 ARG NUMBA_VERSION=""
-ARG CYTHON_VERSION="<3"
+ARG CYTHON_VERSION=""
```

Comment:
If Cython is used to build scikit-learn, note that there are a few performance regressions with Cython 3. So we might want to stick to the latest Cython 2 for the time being.

Reply:
cython is not installed in the final image, so it's not related to the cython one can use to build sklearn later on in a container. Most recent dpctl has merged work that makes it compatible with cython>=3 so I think we can keep that.